src/main/java/view_post/use_case/ViewPostResponseModel.java

```diff
@@ -19,12 +25,26 @@ public class ViewPostResponseModel {
      * @param deadline      The string representation of the deadline of this post.
      */
     public ViewPostResponseModel(String posterEmail, String postBody, String[] postTags, String[] postReplies,
-                                 String deadline) {
+                                 String deadline, String creationDate, String collaborators, int postID, String title) {
         this.posterEmail = posterEmail;
         this.postBody = postBody;
-        this.postTags = postTags;
-        this.postReplies = postReplies;
+        String tags = "";
+        for (String tag : postTags){
+            tags = tags.concat(tag);
+            tags = tags.concat(", ");
+        }
+        tags = tags.substring(0, tags.length() - 2);
+        this.postTags = tags;
+        ArrayList replies = new ArrayList<>();
+        for (String reply : postReplies){
+            replies.add(Integer.valueOf(reply));
+        }
+        this.postReplies = replies;
```

Comment:
Can just initialize `postReplies` as an empty array list and then do the loop after. Removes the need for line 38. Also, as it is, does this not highlight line 38 for using a raw type? It doesn't look like you specified a type for `replies`.